src/main/java/org/thoughtcrime/securesms/preferences/NotificationsPreferenceFragment.java

```diff
@@ -159,7 +161,17 @@ public boolean onPreferenceChange(@NonNull Preference preference, Object newValu
         Ringtone tone = RingtoneManager.getRingtone(getActivity(), value);
 
         if (tone != null) {
-          preference.setSummary(tone.getTitle(getActivity()));
+          String summary;
+          try {
+            summary = tone.getTitle(getActivity());
+          } catch (SecurityException e) {
+            // this could happen in some phones when user selects ringtone from
+            // external storage and later removes the read from external storage permission
+            // and later this method is called from initializeRingtoneSummary()
+            summary = "???";
```

Comment:
maybe mention the situation shown in the comment more precisely - this also has the advantage that it is easier grep'able if it happens gain

```suggestion
            summary = "<no access>";
```